src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2670,5 +2671,260 @@ private interface INegator<T> where T : struct
             public static Vector128<T> NegateIfNeeded(Vector128<T> equals) => ~equals;
             public static Vector256<T> NegateIfNeeded(Vector256<T> equals) => ~equals;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Reverse<T>(ref T elements, nuint length)
+        {
+            Debug.Assert(length > 0);
+            if (!RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+            {
+                if (Unsafe.SizeOf<T>() == sizeof(byte))
+                {
+                    ReverseSIMD(ref Unsafe.As<T, byte>(ref elements), length);
+                    return;
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(char))
+                {
+                    ReverseSIMD(ref Unsafe.As<T, short>(ref elements), length);
+                    return;
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(int))
+                {
+                    ReverseSIMD(ref Unsafe.As<T, int>(ref elements), length);
+                    return;
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(long))
+                {
+                    ReverseSIMD(ref Unsafe.As<T, long>(ref elements), length);
+                    return;
+                }
+            }
+            ReverseScalar(ref elements, length);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void ReverseScalar<T>(ref T elements, nuint length)
+        {
+            if (length <= 1)
+                return;
```

Comment:
nit: While the coding style doesn't require braces for these scenarios, having them is preferred:
```suggestion
            if (length <= 1)
            {
                return;
            }
```